core/src/html/text_format.rs

```diff
@@ -575,11 +575,25 @@ impl FormatSpans {
 
         // quick_xml::Reader requires a [u8] slice, but doesn't actually care about Unicode;
         // this means we can pass the raw buffer in the Latin1 case.
-        let raw_bytes = match html.units() {
-            Units::Bytes(units) => Cow::Borrowed(units),
+        let (raw_bytes, is_raw_latin1) = match html.units() {
+            Units::Bytes(units) => (Cow::Borrowed(units), true),
             // TODO: In principle, we should be able to encode (and later decode)
             // the utf16 units in the [u8] array without discarding losing surrogates.
-            Units::Wide(_) => Cow::Owned(html.to_utf8_lossy().into_owned().into_bytes()),
+            Units::Wide(_) => (
+                Cow::Owned(html.to_utf8_lossy().into_owned().into_bytes()),
+                false,
+            ),
+        };
+
+        // Helper function to decode a byte sequence returned by quick_xml into a WString.
+        // TODO: use Cow<'_, WStr>?
+        let decode_to_wstr = |raw: &[u8]| -> WString {
+            if is_raw_latin1 {
+                WString::from_buf(raw.to_vec())
+            } else {
+                let utf8 = std::str::from_utf8(raw).expect("raw should be valid utf8");
```

Comment:
Please use `str` instead of `std::str`.

Reply:
`from_utf8` isn't a method on `str`, it's a free function in `std::str`.